src/Service.php

```diff
@@ -7,135 +7,287 @@
+			$notFound = true;
+			$section = '';
+			foreach ($this->coveredComponents as $key => $componentDir) {
+				// kontrola jestli dany soubor je soucasti souboru, ktere obsahuji implementaci pro kterou by mel existovat test.
+				if (! Strings::startsWith($_classFile, $componentDir['dir'] . '/')) {
+					continue;
+				}
+				// nalezen, nastavime notFound na false a ukoncime iterovani
+				$notFound = false;
+				$section = $componentDir['section'];
+				break;
+			}
+
+			// pokud jsme dany soubor nenasli, budeme pokracovat dalsi iteraci
+			if ($notFound) {
 				continue;
 			}
 
-			if (! Strings::endsWith($_classFile, 'Presenter.php')) {
+			$mask = $this->config['componentCoverage'][$section]['fileMask'];
+
+			if (! Strings::endsWith($_classFile, $mask)) {
 				continue;
 			}
 
 			require_once $_classFile;
 
 			$_presenterReflection = new \ReflectionClass($_className);
 
+			// nechceme zpracovavat abstraktni tridy -> continue
+			if ($_presenterReflection->isAbstract()) {
+				continue;
+			}
+
+			// maska metody, chceme zpracovat jenom metody ktere maji danou masku
+			$methodMask = null;
+			if (isset($this->config['componentCoverage'][$section]['methodMask'])) {
+				$methodMask = $this->config['componentCoverage'][$section]['methodMask'];
+			}
+
 			foreach ($_presenterReflection->getMethods() as $_presenterMethodReflection) {
-				if (! static::isMethodToTest($_presenterMethodReflection->getName())) {
+				// testy na abstraktni metody nemaji smysl -> continue
+				if ($_presenterMethodReflection->isAbstract()) {
 					continue;
 				}
 
-				$methods[] = $this->getTestClassAndMethod($_presenterReflection->getName(), $_presenterMethodReflection->getName());
+				if (! static::isMethodToTest($_presenterMethodReflection->getName(), $methodMask)) {
+					continue;
+				}
+
+				// potrebujeme ziskat cestu k souboru v mistni slozce, ta odpovida namespace -> smazeme to co je pred namespacem
+				$postionoOfNamespace = stripos($_presenterReflection->getFileName(), str_replace('\\', '/', $_presenterReflection->getName()));
+
+				//bude treba odstranit prvni element z namespace
+				$filePath = substr($_presenterReflection->getFileName(), $postionoOfNamespace);
+				$filePath = substr($filePath, (strpos( $filePath, '/') + 1));
+
+				//plna cesta k testovacimu souboru
+				$testFilePath = realpath($this->config['testDir']). '/' .$filePath."::".$_presenterMethodReflection->getName();
+
+				//vytvareni soupisu metod pro ktere budeme chtit hledat testy
+				$this->methodsToCover[$testFilePath] = $_presenterReflection->getName()."::".$_presenterMethodReflection->getName();
 			}
 		}
-
-		return $methods;
 	}
 
-	protected static function isMethodToTest(string $methodName) : bool
-	{
-		if (Strings::startsWith($methodName, static::$testMethodPrefix)) {
-			return true;
-		}
 
-		return false;
-	}
+	/**
+	 * metoda najde vsechny dostupne testy, ktere mame a ulozi si je
+	 */
+	public function findAvailableTests(): void {
```

Comment:
Jestli to dobře chápu, tak tahle funkce má za úkol naplnit `$this->existingTests`, takže bych ty názvy sjednotil, ať je to jasnější.

Zároveň proč to ty hodnoty rovnou nevrací?

Reply:
A proč je public?

Reply:
Upravil jsem viditelnost na protected

Reply:
Název metody jsem upravil tak, aby odpovídal názvu property do které ukládá data.

Reply:
Přidal jsem návrat nalezených hodnot